keras/models.py

```diff
@@ -1357,8 +1406,10 @@ def fit_generator(self, generator, samples_per_epoch, nb_epoch,
             callbacks: list of callbacks to be called during training.
             validation_data: dictionary mapping input names and outputs names
                 to appropriate numpy arrays to be used as
-                held-out validation data.
-                All arrays should contain the same number of samples.
+                held-out validation data, or a generator yielding such
+                dictionaries. All arrays should contain the same number
+                of samples. If a generator, a single yield will be used
```

Comment:
That makes sense.

However, with such an addition I feel it will be more principled to go ahead and implement `evaluate_generator` and `predict_generator`. I think I will pursue this approach.

Though, perhaps more principled still would be universal generator transparency, i.e. use of generators in basic `fit`, `predict` and `evaluate`, freely mixing data and validation arrays and generators.